### ISSUE TITLE
Allow for currency symbol selection

### ISSUE THREAD

**Issue by wkyoshida** [scribe-org/Scribe-iOS#238]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/scribe-org/Scribe-iOS/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow Scribe-iOS' [Code of Conduct](https://github.com/scribe-org/Scribe-iOS/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

As touched on in this [discussion](https://github.com/scribe-org/Scribe-iOS/issues/229#issuecomment-1281076612), the option to select a currency symbol to use for a keyboard could be a good customization feature.

Some initial thoughts:
- Scribe could quickly prompt the user for the currency symbol during a keyboard setup. This could be done potentially as an addition to the walk-through being worked on in #130, i.e. this issue might be blocked by it.   
- Scribe could allow for separate currency selections per keyboard configured, e.g. a user could select `£` in the German keyboard and `¥` in the Spanish keyboard.
- From [here](https://github.com/scribe-org/Scribe-iOS/issues/229#issuecomment-1284372347), concerning the below:
  > if we do give them the option to choose on data download, then there needs to be a preset value so that we don't lose people in the download funnel 

  Since different currencies are sometimes used in countries where the same language is spoken, to not show "favoritism", Scribe could have the preset value for the currency be a neutral one, e.g. the `£` could be used for Spanish instead of the euro or the peso(s). This is a minor detail, and I'm not even sure if it actually would be welcomed by users (it might not), but just a thought.  
- Apart from the keyboard setup mentioned in the first point, an option within Scribe settings is probably a good idea in case the user wants to change the currency symbol for their existing Scribe keyboard. This is likely a separate issue as a `Settings` page would need to exist.

### Contribution

Willing to help with design and development :rocket:

**Comment by andrewtavis** [comment]
Thanks for making this, @wkyoshida!

- Currently blocked by #16, but will be high on the list to implement once that's done 😊

**Comment by andrewtavis** [comment]
Just assigned to @SaurabhJamadagni 🥳

- Note that this issue is blocked by the menu item issue in #314

**Comment by andrewtavis** [comment]
@vickcoo, some quick notes here for if you'd like to work on this:

- The change will need to happen within `numberKeysPhone` and `symbolKeysPhone` within files like [ENInterfaceVariables.swift](https://github.com/scribe-org/Scribe-iOS/blob/main/Keyboards/LanguageKeyboards/English/ENInterfaceVariables.swift) for each keyboard
    - Looking at it, we also need to make the changes in `numberKeysPad` and `symbolKeysPad` as for smaller iPads we have a just a larger iPhone keyboard (full iPad keyboards are in `letterKeysPadExpanded` and `letterKeysPadExpanded`)
    - In `numberKeysPhone` of the linked file we have `"$"`
    - In `symbolKeysPhone` of the linked file we have `"€", "£", "¥"`
    - Both of the above need to be made into variables now that are then added to the keyboard based on the selection of the menu option in #314
    - We'd have a default one, and then for other options than the default we'd need to change the values of `"€", "£", "¥"`
        - Say the user chooses `"€"`, then we'd want the three in `symbolKeysPhone` to be `"$", "£", "¥"`
    - Maybe a way to do this is have an ordered list of currency symbols per keyboard?
        - For English it would be `"$", "€", "£", "¥"`
        - For German it would be `"€", "$", "£", "¥"`
        - Only the first three appear in `symbolKeysPhone`
        - The user selects the main currency symbol which is then popped/removed from the group of four so it doesn't appear twice
        - If the user selects a symbol that's not there, then it's just the first three that are chosen to be in `symbolKeysPhone`
        - For English: Selecting say rupees `"₹"` would mean that the three remaining would be `"$", "€", "£", `
- From there these variables would then need to be added into the arrays in `numberKeysPhone` and `symbolKeysPhone`
- Then we'd need to connect the work in #314 and update the variables based on the menu selection :)

Let me know if this makes sense and how it sounds! Happy for suggestions or to explain a bit more 😊

**Comment by vickcoo** [comment]
Hi @andrewtavis, I'd like to work on this! I think your idea makes sense, I might need to dive deeper and do more work to figure out what problems need further discussion.

I'm glad you gave me this idea that make it easier for me to quickly understand how to jump into the task ☺️

**Comment by andrewtavis** [comment]
Thanks so much for your willingness to help, @vickcoo! Please let us know if you have further questions :)

**Comment by vickcoo** [comment]
@andrewtavis There's some new progress. Could you review it :)? The English keyboard is done, and I'll update the other keyboards once this idea gets approved.

There was a little question about how can I show you the progress.

**Comment by andrewtavis** [comment]
Closed by #492 ✅ Something to consider here is how we might want to change the potential three extra currency symbols in the future, but I think that should be done solely based on user request as other currency symbols will be available in the alternate characters :)

Really great to have your support there days, @vickcoo! These changes are really improving the quality of the codebase, not just getting the work done for the issues 😊
